packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -72,7 +72,7 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
             const result = jsoncparser.applyEdits(content, edits);
 
             await resource.saveContents(result);
-            this.onDidPreferencesChangedEmitter.fire(undefined);
+            await this.readPreferences();
```

Comment:
We are following the unidirectional data flow, it is updated already via: https://github.com/theia-ide/theia/pull/2707/files#diff-46af11dd5ed879ba4d08304687785b37R56

Reply:
Done, it works!